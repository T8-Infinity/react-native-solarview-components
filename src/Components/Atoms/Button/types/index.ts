import { TouchableOpacityProps } from 'react-native';

export interface IPropsButton extends TouchableOpacityProps {
  title: string | undefined;
  loading?: boolean;
  colorTitle?: string;
  onPress: (() => void) | undefined;
  backgroundColor?: string;
  outline?: boolean;
  disabled?: boolean;
  borderColor?: string;
  testID?: string;
}
