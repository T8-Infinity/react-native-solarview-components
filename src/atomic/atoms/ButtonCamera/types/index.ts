import { TouchableOpacityProps } from 'react-native';

export interface IButtonCameraProps extends TouchableOpacityProps {
  testID?: string;
  icon: string;
  onPress: () => void;
  color: string;
}
