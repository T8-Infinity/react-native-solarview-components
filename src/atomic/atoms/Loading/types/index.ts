import { ActivityIndicatorProps } from 'react-native';

export interface Props extends ActivityIndicatorProps {
  testID?: string;
  size: 'large' | 'small';
  color?: string;
}
