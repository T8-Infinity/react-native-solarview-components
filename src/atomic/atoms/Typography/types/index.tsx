import { fonts } from '@theme/fonts';
import { size } from '@theme/size';

export interface ITextProps {
  text: string;
  fontSize: keyof typeof size;
  color?: string;
  variant: keyof typeof fonts;
}
