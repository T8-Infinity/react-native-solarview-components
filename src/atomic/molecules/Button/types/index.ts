import { TouchableOpacityProps } from 'react-native';

export interface IPropsButton extends TouchableOpacityProps {
  testID?: string;
  title: string;
  loading?: boolean;
  colorTitle?: string;
  onPress: () => void;
  backgroundColor?: string;
  outline?: boolean;
  disabled?: boolean;
  borderColor?: string;
  icon?: string;
}
