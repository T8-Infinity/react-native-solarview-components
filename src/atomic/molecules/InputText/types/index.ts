import { TextInput, TextInputProps } from 'react-native';

export interface InputTextProps extends TextInputProps {
  title: string;
  value: string;
  label?: string;
  icon?: string;
  secret?: boolean;
  touched?: boolean;
  errorMessage?: string;
  onChangeText: (e: any) => void;
  handlePressIcon?: () => void;
  colorTitle?: string;
}

export type Props = InputTextProps & {
  inputRef?: React.RefObject<TextInput>;
};

export interface InputTypesProps {
  colorTitle?: string;
  touched?: boolean;
}
