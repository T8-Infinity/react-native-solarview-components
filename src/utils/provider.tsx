import React from 'react';
import { ThemeProvider } from 'styled-components/native';
import { theme } from '@theme/index';

export const provider: React.FC = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};
