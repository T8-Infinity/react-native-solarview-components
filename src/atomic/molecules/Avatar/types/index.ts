import { IButtonCameraProps } from '@atomic/atoms/ButtonCamera/types';

export interface IAvatarProps {
  testID?: string;
  url: string;
  withLabel?: string;
  type: 'integrator' | 'owner';
  size: 'xs' | 'sm' | 'md' | 'lg' | 'xl';
  color: string;
  icon: string;
}

export type IProps = IAvatarProps & IButtonCameraProps;
