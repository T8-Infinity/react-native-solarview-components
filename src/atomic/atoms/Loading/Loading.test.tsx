import React from 'react';
import { render } from '@testing-library/react-native';
import { provider as wrapper } from '@utils/provider';
import { Props } from './types';

import { Loading } from '.';

describe('Atoms/Loading', () => {
  const elLoading = 'Loading';

  const props: Props = {
    size: 'small',
    color: 'black',
  };

  const setup = () => {
    const utils = render(<Loading {...props} />, {
      wrapper,
    });
    const sut = utils.getByTestId(`${elLoading}`);

    return {
      sut,
      ...utils,
    };
  };

  it(`should be able to render component ${elLoading}`, () => {
    const { sut } = setup();
    expect(sut).toBeTruthy();
  });

  it('should render snapshot', () => {
    const { toJSON } = setup();
    expect(toJSON()).toMatchSnapshot();
  });
});
