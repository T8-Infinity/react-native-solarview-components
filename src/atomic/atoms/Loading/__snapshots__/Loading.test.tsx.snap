// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Loading should render snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
  testID="Loading"
>
  <ActivityIndicator
    color="black"
    size="small"
  />
</View>
`;
